tracer/src/Datadog.Trace/AppSec/EventTrackingSdk.cs

```diff
@@ -17,172 +18,185 @@ namespace Datadog.Trace.AppSec;
+    /// <param name="exists">If the userId associated with the login failure exists</param>
+    /// <param name="metadata">Metadata associated with the login failure</param>
+    [PublicApi]
+    public static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent_Metadata);
+        TrackUserLoginFailureEvent(userId, exists, metadata, Tracer.Instance);
+    }
+
+    internal static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata, Tracer tracer)
+    {
+        if (string.IsNullOrEmpty(userId))
         {
-            TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent);
-            TrackUserLoginFailureEvent(userId, exists, null, Tracer.Instance);
+            ThrowHelper.ThrowArgumentNullException(nameof(userId));
         }
 
-        /// <summary>
-        /// Sets the details of a logon failure on the local root span
-        /// </summary>
-        /// <param name="userId">The userId associated with the login failure</param>
-        /// <param name="exists">If the userId associated with the login failure exists</param>
-        /// <param name="metadata">Metadata associated with the login failure</param>
-        [PublicApi]
-        public static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata)
-        {
-            TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent_Metadata);
-            TrackUserLoginFailureEvent(userId, exists, metadata, Tracer.Instance);
-        }
+        var span = tracer?.ActiveScope?.Span;
 
-        internal static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata, Tracer tracer)
+        if (span is null)
         {
-            if (string.IsNullOrEmpty(userId))
-            {
-                ThrowHelper.ThrowArgumentNullException(nameof(userId));
-            }
-
-            var span = tracer?.ActiveScope?.Span;
-
-            if (span is null)
-            {
-                ThrowHelper.ThrowException("Can't create a tracking event with no active span");
-            }
+            ThrowHelper.ThrowException("Can't create a tracking event with no active span");
+        }
 
-            var setTag = TaggingUtils.GetSpanSetter(span);
+        var setTag = TaggingUtils.GetSpanSetter(span, out var spanInternal);
 
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureTrack, "true");
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureSdkSource, "true");
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureUserId, userId);
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureUserExists, exists ? "true" : "false");
+        setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureTrack, "true");
```

Comment:
NIT: While it was already there before this PR, maybe we could consider using constants for "true" and "false" to improve readability and decrease allocations (although the compiler might optimize that, but with this change we would be certain about this optimization)?

Using something like:

public static class Constants
{
    public static readonly string TrueValue = "true";
    public static readonly string FalseValue = "false";
}

Reply:
Thanks for the idea, I'll do this in the next PR, fixing user id collection!

Reply:
Use actual constants though, not `static readonly` 😄